libenv/sysinfo.c

```diff
@@ -2158,8 +2158,8 @@ static int Linux_Suse_Version(EvalContext *ctx)
         }
         else
         {
-            sscanf(strversion, "VERSION = %s", strmajor);
-            sscanf(strpatch, "PATCHLEVEL = %s", strminor);
+            sscanf(strversion, "VERSION = %255s", strmajor);
+            sscanf(strpatch, "PATCHLEVEL = %255s", strminor);
```

Comment:
And here. And all places below.

I know it's ugly, but in C, safety is more important :)